src/main/java/com/zombiecastlerush/building/Puzzle.java

```diff
@@ -0,0 +1,82 @@
+package com.zombiecastlerush.building;
+
+import java.util.*;
+
+class Puzzle extends Challenge{
+    private String name;
+    private int attempts;
+    private final int max_attempts=3;
+    private Inventory puzzInventory;
+    private HashMap<String, String> questionAnswers = new HashMap<>();
+
+    //Constructor
+    public Puzzle(String name,String description){
+        super(description);
+        this.setName(name);
+        this.attempts=0;
+    }
+
+    //Getters and setters
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getAttempts() {
+        return attempts;
+    }
+
+    public Inventory getPuzzInventory() {
+        return puzzInventory;
+    }
+
+    //Methods
+    //initializes the questionAnswers Map of Puzzle with question as the key and solution as the value
+    public void setPuzzle(String puzzleQuestion, String solution) {
+        questionAnswers.put(puzzleQuestion,solution);
+    }
+
+    public void presentPuzzle(){
+        System.out.println(getDescription());
+    }
+
+    public void attemptPuzzle(){
+        List<String> listOfQuestions = new ArrayList<>(questionAnswers.keySet());
+
+        if(getAttempts()<max_attempts){
+            attempts += 1;
+            Scanner myScanner = new Scanner(System.in);
```

Comment:
Perhaps it would be better to leave all the Scanner stuff to the Prompter class - saving us the trouble of going through and hunting through odd bugs that stem of our delegating control of the application to Scanner lines all throughout our repo. You could instead just take a String input that is compared and delivers a boolean back, or an int for tries left. Or even an Answer class?

Reply:
That is the purpose of our Prompter. whenever we need input from our user or player, we should call Prompter.